vcd/resource_vcd_org_vdc_test.go

```diff
@@ -0,0 +1,196 @@
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVdcDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdVdcExists("vcd_org_vdc."+TestAccVcdVdc, &vdc),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "name", TestAccVcdVdc),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "org", testConfig.VCD.Org),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "allocation_model", "ReservationPool"),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "network_pool_name", testConfig.VCD.ProviderVdc.NetworkPool),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "provider_vdc_name", testConfig.VCD.ProviderVdc.Name),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "enabled", "true"),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "enable_thin_provisioning", "true"),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "enable_fast_provisioning", "true"),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "delete_force", "true"),
+					resource.TestCheckResourceAttr(
+						"vcd_org_vdc."+TestAccVcdVdc, "delete_recursive", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckVcdVdcExists(name string, vdc *govcd.Vdc) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("not found: %s", name)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no VDC ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*VCDClient)
+
+		adminOrg, err := conn.GetAdminOrg(testConfig.VCD.Org)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingOrg, testConfig.VCD.Org+" and error: "+err.Error())
+		}
+
+		newVdc, err := adminOrg.GetVdcByName(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("vdc %s does not exist (%#v)", rs.Primary.ID, newVdc)
+		}
+
+		vdc = &newVdc
+		return nil
+	}
+}
+
+func testAccCheckVdcDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*VCDClient)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "vcd_vdc" && rs.Primary.Attributes["name"] != TestAccVcdVdc {
+			continue
+		}
+
+		adminOrg, err := conn.GetAdminOrg(testConfig.VCD.Org)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingOrg, testConfig.VCD.Org+" and error: "+err.Error())
+		}
+
+		vdc, err := adminOrg.GetVdcByName(rs.Primary.ID)
+
+		if vdc != (govcd.Vdc{}) {
+			return fmt.Errorf("vdc %s still exists", rs.Primary.ID)
+		}
+		if err != nil {
+			return fmt.Errorf("vdc %s still exists or other error: %#v", rs.Primary.ID, err)
+		}
+
+	}
+
+	return nil
+}
+
+func init() {
+	testingTags["vdc"] = "resource_vcd_vdc_test.go"
```

Comment:
It should be `resource_vcd_org_vdc_test.go`

Reply:
Fixed